enclave/core/tracee.c

```diff
@@ -53,28 +53,42 @@ const char* get_filename_from_path(const char* path)
 ** Handle the OE_ECALL_LOG_INIT from host and initialize SDK logging
 ** configuration
 **
+** If enclave_path = NULL, then only the active log_level is modified.
+**
 **==============================================================================
 */
 
 void oe_log_init_ecall(const char* enclave_path, uint32_t log_level)
 {
+    oe_result_t result = OE_OK;
     const char* filename;
+    bool locked = false;
 
     // Returning OE_UNSUPPORTED means that the logging.edl is not properly
     // imported. Do not perform the initialization in this case.
     if (oe_log_is_supported_ocall() == OE_UNSUPPORTED)
         return;
 
+    OE_CHECK_NO_TRACE(oe_mutex_lock(&_log_lock));
+    locked = true;
```

Comment:
why is this needed?

Reply:
This is to make sure that this log level change does not happen simultaneously when a log is being logged. Similar to oe_set_host_log_level()

Reply:
I was referring to that is there any particular reason that the extra variable needed? For example, in `oe_set_host_log_level`, `oe_mutex_unlock` is called inside the if bracket.

Reply:
I removed the 1locked` variable. It is not needed anymore.